tests/platform_tests/fwutil/test_fwutil.py

```diff
@@ -0,0 +1,119 @@
+    found = False
+    pattern = re.compile(r'.*Error: Invalid value for "<fw_path>"*.')
+    for line in out['stderr_lines']:
+        if pattern.match(line):
+            found = True
+            break
+    assert found
+
+def test_fwutil_update_current(duthost, localhost, pdu_controller, fw_pkg, random_component):
+    assert call_fwutil(duthost,
+            localhost,
+            pdu_controller,
+            fw_pkg, 
+            component=random_component)
+
+def test_fwutil_update_next(duthost, localhost, pdu_controller, fw_pkg, random_component, next_image):
+    assert call_fwutil(duthost, 
+            localhost,
+            pdu_controller,
+            fw_pkg, 
+            component=random_component, 
+            next_image=next_image)
+
+def test_fwutil_update_bad_config(duthost, fw_pkg, random_component):
+    versions = show_firmware(duthost)
+    chassis = versions["chassis"].keys()[0] # Only one chassis
+
+    # Test fwutil update with config file without chassis section
+    
+    with open("platform_components.json", "w") as f:
+        json.dump({}, f, indent=4)
+    upload_platform(duthost, {})
+    found = False
+    out = duthost.command("fwutil update chassis component {} fw -y".format(random_component), module_ignore_errors=True)
+    pattern = re.compile(r'.*Error: Failed to parse "platform_components.json": invalid platform schema*.')
+    for line in out["stdout_lines"]:
+        if pattern.match(line):
+            found = True
+            break
+    assert found
+
+    # Test fwutil update with config file without component section
+    with open("platform_components.json", "w") as f:
+        json.dump({"chassis":{chassis:{}}}, f, indent=4)
+    upload_platform(duthost, {})
+    found = False
+    out = duthost.command("fwutil update chassis component {} fw -y".format(random_component), module_ignore_errors=True)
+    pattern = re.compile(r'.*Error: Failed to parse "platform_components.json": invalid chassis schema*.')
+    for line in out["stdout_lines"]:
+        if pattern.match(line):
+            found = True
+            break
+    assert found
+
+    # Test fwutil update with config file with version of type dict
+    with open("platform_components.json", "w") as f:
+        json.dump({"chassis":{chassis:{"component":{random_component:{"version":{"version":"ver"}}}}}}
+                , f, indent=4)
+    upload_platform(duthost, {})
+    found = False
+    out = duthost.command("fwutil update chassis component {} fw -y".format(random_component), module_ignore_errors=True)
+    pattern = re.compile(r'.*Error: Failed to parse "platform_components.json": invalid component schema*.')
+    for line in out["stdout_lines"]:
+        if pattern.match(line):
+            found = True
+            break
+    assert found
+
+@pytest.mark.skip(reason="Command not yet merged into sonic-utilites")
+@pytest.mark.parametrize("reboot_type", ["none", "warm", "fast", "cold", "power off"])
```

Comment:
This test is a placeholder right now because the `fwutil update all` command is not yet merged to master, I would propose examining what would be best for this once its merged in and we can start running it. 

The code is simply here to show high level design and has a hardcoded skip.

Reply:
I do agree we may want that to eventually be a randomization.